Optik/Classes/ImageViewerDelegate.swift

```diff
@@ -30,4 +30,20 @@ public protocol ImageViewerDelegate: class {
      */
     func imageViewerDidDisplayImage(at index: Int)
     
+    /**
+     Tells when the user touch in the like button
+     
+     - parameter index: Index of the image.
+    */
+    func didTouchLikeButton(at index: Int)
+ 
+    
+    /**
+      Ask delegate for the image that should appear in the button when the image appears.
+      - parameter index: index of the image
+     
+      - return: UIImage
+    */
+    func getImageFromButton(at index: Int) -> UIImage
```

Comment:
Are we talking about the like button? The other method up there is called didTouchLikeButton, here it's just Button. Should make things consistent..